app/models/manageiq/providers/amazon/inventory/collector/cloud_manager.rb

```diff
@@ -49,17 +49,23 @@ def referenced_images
 
     multi_query(extra_image_references) do |refs|
       hash_collection.new(
-        aws_ec2.client.describe_images(:filters => [{:name => 'image-id', :values => refs}]).images
+        aws_ec2.client.describe_images(:filters => [{:name => 'image-id', :values => refs}]).flat_map(&:images)
       ).all
     end
   end
 
   def stacks
-    hash_collection.new(aws_cloud_formation.client.describe_stacks[:stacks])
+    hash_collection.new(aws_cloud_formation.client.describe_stacks.flat_map(&:stacks))
   end
 
   def stack_resources(stack_name)
-    stack_resources = aws_cloud_formation.client.list_stack_resources(:stack_name => stack_name).try(:stack_resource_summaries)
+    stack_resources = aws_cloud_formation.client.list_stack_resources(:stack_name => stack_name)
+
+    if stack_resources.respond_to?(:stack_resource_summaries)
```

Comment:
Is this right?  The stack_resources is a collection, whereas `stack_resource_summaries` would be a method on the indivudual items.

Reply:
Or does `list_stack_resources` work different from `describe_*`?

Reply:
Honestly I wasn't sure, and was going to ask about this. At worst it gives you the same result:

```
stack_name = "lucy-orch1"

irb(main):028:0> stack_resources = cf.client.list_stack_resources(:stack_name => name).stack_resource_summaries
=> [#<struct Aws::CloudFormation::Types::StackResourceSummary logical_resource_id="Ec2Instance", physical_resource_id=nil, resource_type="AWS::EC2::Instance", last_updated_timestamp=2019-11-21 21:52:02 UTC, resource_status="DELETE_COMPLETE", resource_status_reason=nil, drift_information=#<struct Aws::CloudFormation::Types::StackResourceDriftInformationSummary stack_resource_drift_status="NOT_CHECKED", last_check_timestamp=nil>>]

irb(main):029:0> stack_resources = cf.client.list_stack_resources(:stack_name => name).flat_map(&:stack_resource_summaries)
=> [#<struct Aws::CloudFormation::Types::StackResourceSummary logical_resource_id="Ec2Instance", physical_resource_id=nil, resource_type="AWS::EC2::Instance", last_updated_timestamp=2019-11-21 21:52:02 UTC, resource_status="DELETE_COMPLETE", resource_status_reason=nil, drift_information=#<struct Aws::CloudFormation::Types::StackResourceDriftInformationSummary stack_resource_drift_status="NOT_CHECKED", last_check_timestamp=nil>>]
```